### ISSUE TITLE
Some way to refer to the models (and data structures) in the package?

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#448]
In PR #447 I suggested adding documentation like:

```
#' @param data An object with class corresponding to an implemented model.
#' The currently available models classes are `epidist_latent_model` and
#' `epidist_naive_model`.
```

I think this is fine for now but really we don't want to have to be updating documenation like this when we add a new model.

Instead, I think perhaps we could have some kind of data object or function in the package which lists the available models. Then in documentation we say that you can call `epidist_model_classes()` to look at them. Or for the data ones we can say that they can call `epidist_data_classes()`.

What do you think @seabbs? I think this is something you've probably encountered before.

**Comment by athowes** [comment]
Here's a possible design: 

```r
> epidist_models <- function() {
+   all_objects <- ls(asNamespace("epidist"))
+   model_functions <- grep("^as_epidist_.*_model$", all_objects, value = TRUE)
+   model_names <- sub("^as_epidist_([a-z_]+)$", "\\1", model_functions)
+   unique(model_names)
+ }
> epidist_models()
[1] "latent_model"   "marginal_model" "naive_model"   
```

Could do similar for data classes.

**Comment by seabbs** [comment]
Yes could do. Maybe in the first instance though we jus want to have aim for a vignette that lists the models in a single place and then point to that?

**Comment by seabbs** [comment]
it is I think using vignettes("name of vignette")

**Comment by athowes** [comment]
> Maybe in the first instance though we jus want to have aim for a vignette that lists the models in a single place and then point to that?

Creating and upkeeping a vignette which lists the models seems to be more work to do than the solution above and also requires more maintanence.

Eventually we want the vignette explaining the models to be up, and perhaps at that point we could switch to pointing to it.

However, that vignette might be quite "mathy" in which case I think allowing users to just get a simple list of the models in the package is still valuable functionality (rather than having to read a load of notation before figuring out which models exist).

**Comment by athowes** [comment]
> to list the methods each model uses so I think we can't get away from that

Isn't this the functions reference page?

**Comment by seabbs** [comment]
Well maybe but there is too much noise to be useful there I think (due to all the methods per function)

**Comment by seabbs** [comment]
I think going to close this as not planned and rely on docs to manage. Happy to reopen in. the future/ if there are strong feelings.
